src/Symfony/Component/Cache/Adapter/CouchbaseBucketAdapter.php

```diff
@@ -0,0 +1,226 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Cache\Adapter;
+
+use Couchbase\Bucket;
+use Symfony\Component\Cache\Exception\CacheException;
+use Symfony\Component\Cache\Exception\InvalidArgumentException;
+use Symfony\Component\Cache\Marshaller\DefaultMarshaller;
+use Symfony\Component\Cache\Marshaller\MarshallerInterface;
+
+/**
+ * Class CouchbaseBucketAdapter.
+ */
+class CouchbaseBucketAdapter extends AbstractAdapter
```

Comment:
why not just CouchbaseAdapter?

Reply:
Because couchbase uses a connection against a bucket, since I can have different buckets of different types.
Actually the connection that is used is the connection against the bucket.

Reply:
final class...